borbskeeper/Features/Profile/ProfileViewController.m

```diff
@@ -0,0 +1,162 @@
+//
+//  ProfileViewController.m
+//  borbskeeper
+//
+//  Created by rodrigoandrade on 7/25/19.
+//  Copyright © 2019 juliapark628. All rights reserved.
+//
+
+#import "ProfileViewController.h"
+#import "UIImageView+AFNetworking.h"
+#import "BorbParseManager.h"
+#import "Task.h"
+#import "CompletedTasksListInfiniteScrollView.h"
+#import "TaskCell.h"
+
+@interface ProfileViewController ()<UIImagePickerControllerDelegate, UINavigationControllerDelegate, InfiniteScrollDelegate>
+
+@property (weak, nonatomic) IBOutlet UIImageView *profilePicture;
+@property (weak, nonatomic) IBOutlet UILabel *userName;
+@property (weak, nonatomic) IBOutlet CompletedTasksListInfiniteScrollView *completedTasksListInfiniteScrollView;
+@property (weak, nonatomic) IBOutlet UIActivityIndicatorView *activityIndicator;
+
+@property (strong, nonatomic) NSMutableArray *completeTaskList;
+@property (strong, nonatomic) NSDate *latestDate;
+@property (strong, nonatomic) UIImage *originalImage;
+@property (strong, nonatomic) UIImage *editedImage;
+@property (nonatomic, strong) UIRefreshControl *refreshControl;
+
+@end
+
+@implementation ProfileViewController
+
+static NSString *const USER_PROF_PIC_KEY = @"profilePicture";
+static NSString *const COMPLETE_TASK_TABLE_VIEW_CELL_ID = @"CompletedTaskCell";
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.completedTasksListInfiniteScrollView.infiniteScrollDelegate = self;
+    [self setupProfile];
+    [self fetchData];
+    [self refreshCompleteTaskList];
+}
+
+- (void)setupProfile{
+    NSString *userName = User.currentUser.username;
+    self.userName.text = userName;
+    [self loadUserProfilePicture];
+}
+
+-(void)loadUserProfilePicture{
+    PFFileObject *PFObjectProfileImage = [User currentUser][USER_PROF_PIC_KEY];
+    NSURL *profileImageURL = [NSURL URLWithString:PFObjectProfileImage.url];
+    self.profilePicture.image = nil;
+    [self.profilePicture setImageWithURL:profileImageURL];
+}
+
+- (UIImage *)resizeImage:(UIImage *)image withSize:(CGSize)size {
+    UIImageView *resizeImageView = [[UIImageView alloc] initWithFrame:CGRectMake(0, 0, size.width, size.height)];
+    resizeImageView.contentMode = UIViewContentModeScaleAspectFill;
+    resizeImageView.image = image;
+    
+    UIGraphicsBeginImageContext(size);
+    [resizeImageView.layer renderInContext:UIGraphicsGetCurrentContext()];
+    UIImage *newImage = UIGraphicsGetImageFromCurrentImageContext();
+    UIGraphicsEndImageContext();
+    return newImage;
+}
+
+- (void)imagePickerController:(UIImagePickerController *)picker didFinishPickingMediaWithInfo:(NSDictionary<NSString *,id> *)info {
```

Comment:
is this the only place where we are using pickercontroller?

Reply:
I mean, is this the only view controller?

Reply:
Currently yes, but if we end up implementing photo-based task verification it might be used again.